common/src/main/java/org/opensearch/ml/common/transport/execute/MLExecuteTaskRequest.java

```diff
@@ -31,21 +33,25 @@
 @ToString
 public class MLExecuteTaskRequest extends ActionRequest {
 
+    FunctionName functionName;
```

Comment:
Is this FunctionName field the key part that triggered the bug? I see that this triggered most of the required changes in the PR.

Reply:
Yes, we need `functionName` to tell which class to initialize for this algorithm.